### ISSUE TITLE
create separate captions for each text decoder for sdxl training

### ISSUE THREAD

**Issue by mio-nyan** [kohya-ss/sd-scripts#938]
After some debugging I found that captions and tags from the config metafile are just merged togheter.
`caption = caption + ", " + tags` train_util.py line 1564

To my understanding (and a lot of try & error) the quality of sdxl generated images are a lot better, when using a different kind of syntax for each encoder. Since the training script already processes captions and tags separately, it might be possibly to keep them separately until input_ids creation?  So that each text encoder has it's own captions. 

When prompting for both encoders at the same time, I think you would also use either captions or tags. so by merging them togheter during training the outcome will be more unpredictable.

Moreover it would be nice to also have separate prompts during sample image generation in order to test this (and also the different learning rates for each text encoder!)

I'm currently trying to implement it, let's see how far I get xD

**Comment by dill-shower** [comment]
+1

**Comment by dill-shower** [comment]
Duplicate https://github.com/kohya-ss/sd-scripts/issues/781

**Comment by mio-nyan** [comment]
Ahh, big sorry! Why didn't I see that!

**Comment by mio-nyan** [comment]
To be honest, I don't know exactly how to integrate the whole thing properly.
Since the training script already processes captions and tags separately, I kept them separately until input_ids creation instead of merging them earlier. This alone wouldn't be that much of a change.

It's more about the concept. Atm there are things like tag shuffling / drop captions / ... which adds additional randomness to the whole training process.
Moreover captioning is mixed throughout the code for 1.5, 2.x and sdxl. So it would make sense to split the sdxl from the rest.

For the 2 extra files: I just copied 
merge_captions_to_metadata.py
merge_dd_tags_to_metadata.py
and changed it to create a metafile with captionG and captionL which is easier to understand than caption and tags
